### PR TITLE
feat: Added Creation Date col in Rooms Table in Admin Panel

### PR DESCRIPTION
**Pull request by heet434**
## Proposed changes (including videos or screenshots)
Added a column that displays the room creation date. 
Screenshot 2024-07-02 at 2 16 14 PM

Also added a ts field to retrieve the creation date from the database.

## Issue(s)

Here's the link: https://github.com/RocketChat/feature-requests/issues/893

## Steps to test or reproduce

Open the Rooms section in Admin Panel. Need for a creation date column there.

## Further comments

https://rocketchat.atlassian.net/browse/CONN-293

### CODE REVIEW

**Comment by hugocostadev** [comment]
> Hi @hugocostadev , I have resolved the typecheck and lint errors, and used the useFormatDate function for formatting the date, but I can't find out as to what exactly is failing the unit test in @rocket.chat/license:testunit . Could you please help me out here?

Hey @heet434 I'll review it again! Thanks! 

Related to the unit test, don't worry... it's expected, we are going to fix this internally

**Comment by heet434** [comment]
> > Hi @hugocostadev , I have resolved the typecheck and lint errors, and used the useFormatDate function for formatting the date, but I can't find out as to what exactly is failing the unit test in @rocket.chat/license:testunit . Could you please help me out here?
> 
> Hey @heet434 I'll review it again! Thanks!
> 
> Related to the unit test, don't worry... it's expected, we are going to fix this internally

@hugocostadev  Thank you! I'll add the commits that resolve the prev issue.

**Review by hugocostadev — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/admin/rooms/RoomsTable.tsx [hunk 1]
```diff
@@ -132,7 +135,7 @@ const RoomsTable = ({ reload }: { reload: MutableRefObject<() => void> }): React
 						<GenericTableHeader>{headers}</GenericTableHeader>
 						<GenericTableBody>
 							{data.rooms?.map((room) => (
-								<RoomRow key={room._id} room={room} />
+								<RoomRow key={room._id} room={{ ...room, ts: room.ts ? new Date(room.ts) : undefined }} />
```
[the remark this answers is not in the document]
Comment (heet434) [reply apps/meteor/client/views/admin/rooms/RoomsTable.tsx:138]: Hi, thank you for the review. If I remove the constructor, typescipt starts throwing type errors as the 'ts' is taken in as a string initially, and if I use 'Date | string' as its type when defining the type in IRoom.ts, there's again another type error in another file. Is there a quick fix for this?
Screenshot 2024-07-03 at 12 17 29 AM
<br/>
And here's the error listed after running yarn turbo run typecheck.
Screenshot 2024-07-03 at 12 58 05 AM

[the remark this answers is not in the document]
Comment (hugocostadev) [reply apps/meteor/client/views/admin/rooms/RoomsTable.tsx:138]: Yep @heet434 indeed this TS error appears...
This happens due to the JSON deserialization, the JS transforms type Date to String in the process... 

I'll do a commit that simulates this and bypass this error

[the remark this answers is not in the document]
Comment (heet434) [reply apps/meteor/client/views/admin/rooms/RoomsTable.tsx]: Okay thank you @hugocostadev , is there anything else I should change or add ?

[the remark this answers is not in the document]
Comment (hugocostadev) [reply apps/meteor/client/views/admin/rooms/RoomsTable.tsx]: No need! Thanks

**Review by hugocostadev — DISMISSED** [review]

**Review by hugocostadev — APPROVED** [review]

**Review by tassoevan — APPROVED** [review]

**Pull request merged**